src/v2i-hub/PedestrianPlugin/src/FLIRWebSockAsyncClnSession.cpp

```diff
@@ -227,80 +239,87 @@ namespace PedestrianPlugin
             if (!it.second.get_child("iD").data().empty()) 
             {
                 id = std::stoi(it.second.get_child("iD").data());
-
-                // Convert the id to 4 octet string
-                std::stringstream idstream;
-                idstream << std::hex << id;
-                idResult = idstream.str();
-                int str_length_diff = 8 - static_cast<int>(idResult.length());
-                idResult.append(str_length_diff, '0');
-
-                // Need to convert ID for SDSM
-                // Map the value to be less than 65535
-                auto mappedID = id % 65535;
-                objID = std::to_string(mappedID);
+                if (id > 65535)
+                {
+                    throw std::out_of_range("ID out of range");
+                }
+                objID = std::to_string(id);
             }
 
-            // Parse latitude
-            if (!it.second.get_child("latitude").data().empty())
+            
+            // Convert configured lat/lon to J2735 lat/lon format
+            int latInt = std::round(fLIRLat_* 10e6);
+            lat = std::to_string(latInt);
+            // Convert configured lat/lon to J2735 lat/lon format
+            int lonInt = std::round(fLIRLon_* 10e6);
+            lon = std::to_string(lonInt);
+            
+            if (!it.second.get_child("x").data().empty())
             {
+                int xOffsetInt = std::round(std::stod(it.second.get_child("x").data())*10);
                 // Convert received lat/lon to J2735 lat/lon format
-                lat = it.second.get_child("latitude").data();
-                lat.erase(std::remove(lat.begin(), lat.end(), '.'), lat.end());
-                lat.pop_back();
+                xOffset = std::to_string(xOffsetInt);
```

Comment:
Not sure if this comment apply to the following line of code. The same comment for y offset